### ISSUE TITLE
Mapping Composition does not work for @SubclassMapping

### ISSUE THREAD

**Issue by bwgjoseph** [mapstruct/mapstruct#3174]
### Expected behavior

I'm trying to use [mapping-composition](https://mapstruct.org/documentation/stable/reference/html/#mapping-composition) for `@SubclassMapping`

Coming from https://github.com/mapstruct/mapstruct/issues/2528#issuecomment-1435735687 where @Zegveld mentioned that it should be supported.

### Actual behavior

When creating the `meta-annotation`, it complains the following

```log
The annotation @SubclassMapping cannot be repeated at this location since its container annotation type @SubclassMappings is disallowed at this location
```

![image](https://user-images.githubusercontent.com/37484538/221375730-615c2570-1258-4e36-9cec-7c60b3e84df7.png)

![image](https://user-images.githubusercontent.com/37484538/221375743-044818c4-5f71-4d1e-91bd-57b42da3bac6.png)

### Steps to reproduce the problem

I have the following setup

```java
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME)
@JsonSubTypes({
    @Type(value = HomeAddressRequestDto.class, name = "home"),
    @Type(value = OfficeAddressRequestDto.class, name = "office")
})
public interface AddressRequestDto {}

@Builder
@JsonTypeName("home")
public record HomeAddressRequestDto(
        String id,
        String street,
        String postalCode,
        String unit,
        String createdBy,
        String createdAt,
        Country country,
        Point location)
        implements AddressRequestDto {}

@Builder
@JsonTypeName("office")
public record OfficeAddressRequestDto(String building, String street, String postalCode, String unit, String createdBy, String createdAt) implements AddressRequestDto {}

@Mapper(
        componentModel = MappingConstants.ComponentModel.SPRING,
        subclassExhaustiveStrategy = SubclassExhaustiveStrategy.RUNTIME_EXCEPTION
)
public interface AddressMapper {
    @SubclassMapping(source = HomeAddressRequestDto.class, target = HomeAddressDO.class)
    @SubclassMapping(source = OfficeAddressRequestDto.class, target = OfficeAddressDO.class)
    AddressDO toDomainObject(AddressRequestDto addressDto);
```

Which I am trying to use `Mapping Composition`

```java
@Retention(RetentionPolicy.CLASS)
@SubclassMapping(source = HomeAddressRequestDto.class, target = HomeAddressDO.class)
@SubclassMapping(source = OfficeAddressRequestDto.class, target = OfficeAddressDO.class)
public @interface CombinedSubclassMapping {}
```

Then change the mapper class

```java
public interface AddressMapper {
    @CombinedSubclassMapping
    AddressDO toDomainObject(AddressRequestDto addressDto);
```

### MapStruct Version

1.5.3.Final

**Comment by Zegveld** [comment]
This indeed seems to be a bug with the `@SubclassMappings` (plural) annotation, where the meta annotation category probably is missing. I'll work on a PR with a fix for this.

**Comment by bwgjoseph** [comment]
Hi @filiphr, I see this being closed and there's a new release of 1.5.5 which should include this fix? Even though this is tagged to 1.6.0 milestone.

I tried on 1.5.5, but it doesn't seem to be fixed. Thanks

**Comment by filiphr** [comment]
@bwgjoseph it doesn't work on 1.5.5. because it isn't part of that release. This is closed because it has been done on main. As you can see the target milestone is 1.6.0. So it will be part of that release.
